extensions/wikia/TaxonomyArticleHooks/TaxonomyArticleHooks.class.php

```diff
@@ -0,0 +1,43 @@
+<?php
+
+use Wikia\Rabbit\ConnectionBase;
+
+/**
+ * @author Lore team
+ *
+ * Send plaintext versions of articles to RabbitMQ as they are edited on Fandom
+ */
+
+class TaxonomyArticleHooks {
+	const ROUTING_KEY = "taxonomy.article-edits";
+
+	protected static $rabbitConnection;
+
+	static public function onArticleEditUpdates( \WikiPage $page, $editInfo ) {
+		global $wgCityId;
+
+		// We currently do not need wikitext for the classifier, but it may be useful
+		//  to have in the future. Uncomment this line to include wikitext in the message
+		$message = [
+			'wikiId' => $wgCityId,
+			'articleId' => $page->getId(),
+			'articleTitle' => $page->getTitle()->getText(),
+			// 'articleWikitext' => $page->getText(),
+			'articlePlaintext' => html_entity_decode( strip_tags( $editInfo->output->getText() ), ENT_COMPAT, 'UTF-8' )
```

Comment:
I wonder if that is not an overkill for RabbitMQ. Some pages tend to be quite complex and putting the whole content into AMPQ message can cause problems (memory/cpu usage spikes). I would prefer to fetch content asynchronously on the consumer side and not pump all the data through the RabbitMQ

Reply:
It's something I've addressed in the PR. Rabbit can handle messages up to around half a gig, and even complex articles should only be around a few kb. We'll be monitoring the queue and have discussed this implementation with Frank ahead of time. Since messages that go in should be immediately consumed I can't see any immediate problems here.

Reply:
I'm mildly optimistic about that ;-) It's the volume of those messages that worries me. 

Either way remember to put alerts in place like this one: https://github.com/Wikia/chef-repo/blob/master/cookbooks/kapacitor-18/attributes/platform-core.rb#L33-L34 before you go live on prod.